docs/models_dev_guide.md

```diff
@@ -189,38 +190,50 @@ automatically load a model given only its `name`. See the
 
 #### Publishing a new model
 
-To publish a new model, you can follow the general receipt:
+By default, all ETA models are stored in a Google Drive folder with permissions
+set to be publicly readable. You can publish a new model to this public
+registry as follows:
 
 ```py
 #
-# Full workflow for publishing a new model
+# Publishing a new public ETA model
 #
+# Assumes the model has been uploaded to Google Drive by an ETA administrator
+# and you have the ID of the model in Google Drive
+#
+
+# The name for your model
+name = "your-model@1.0"
 
-# Recommend paths for the given model by looking for older versions
-# of the model in the model manifests
-base_filename, models_dir = etam.recommend_paths_for_model(name)
+# The ID of your model in Google Drive
+google_drive_id = "XXXXXXXX"
 
-# Perform a dry run of the model registration to check for any errors
-# before uploading the model to the cloud
-etam.register_model_dry_run(name, base_filename, models_dir)
+# A short description of your model
+description = "a short description of your model"
 
-# Upload the file to remote storage
-# Then instantiate the relevant ModelManager describing its location
-manager = ...
+#
+# The base filename (no version information, which is added automatically) and
+# models directory, respectively, to use when downloading the model. If you
+# are uploading a new version of an existing model, these arguments can be set
+# to None and the same values are inherited from the most recent version of the
+# model
+#
+base_filename = "your-model-weights.npz"
+models_dir = "/path/to/models/dir"
```

Comment:
As a reader of this doc, I see `models_dir`, I read the paragraph above `to use when downloading the model` and I think these thoughts:
1.  Isn't that models directory configured by all of eta?  Why do I need to specify it?  
2.  How will I know the root path to where people want to store their models?
3.  Is this also where it is stored in the eta public models repository?
4.  How do I tell the system where the current weights are?  Is this the same as `models_dir`?
Please clarify (if only to avoid you needing to answer these for everyone who uses it).

Reply:
1. I updated the code to fallback to the first directory on the model search path if necessary. This was definitely missing before.
4. This process doesn't involve the actual model file because it must have been previously uploaded to Google Drive (we shouldn't allow third-parties the ability to do that themselves)

For the rest, I added some more content to the dev guide. Hopefully it makes sense.